lighthouse-cli/test/smokehouse/smokehouse.js

```diff
@@ -97,9 +97,61 @@ function runLighthouse(url, configPath) {
 }
 
 /**
+ * Walk down expected result, comparing to actual result. If a difference is found,
+ * the path to the difference is returned, along with the expected primitive value
+ * and the value actually found at that location. If no difference is found, returns
+ * null.
+ *
+ * Only checks own enumerable properties, not object prototypes, and will loop
+ * until the stack is exhausted, so works best with simple objects (e.g. parsed JSON).
+ * @param {string} path
+ * @param {*} actual
+ * @param {*} expected
+ * @return {({path: string, actual: *, expected: *}|null)}
+ */
+function findDifference(path, actual, expected) {
+  // Strict equality check, plus NaN equivalence.
+  if (Object.is(actual, expected)) {
+    return null;
+  }
+
+  // if they aren't both an object, reject.
```

Comment:
Maybe something other than "reject". ...Can't create diff if actual and expected are both not objects.